include/diaspora.php

```diff
@@ -828,7 +848,7 @@ private static function allowed_contact_by_handle($importer, $handle, $is_commen
 	 * @return int|bool message id if the message already was stored into the system - or false.
 	 */
 	private static function message_exists($uid, $guid) {
-		$r = q("SELECT `id` FROM `item` WHERE `uid` = %d AND `guid` = '%s' LIMIT 1",
+		$r = q("SELECT `guid` FROM `item` WHERE `uid` = %d AND `guid` = '%s' LIMIT 1",
 			intval($uid),
 			dbesc($guid)
 		);
```

Comment:
I can't comment below, but please add `dbm::is_result()` line 856 and change `id` to `guid` line 858. Are you sure it's not affecting the functions calling `message_exists`?